File: app/src/main/java/com/jbrunton/mymovies/api/resources/MoviesCollection.java
package com.jbrunton.mymovies.api.resources;

import com.jbrunton.entities.Configuration;
import com.jbrunton.entities.Movie;

import java.util.List;

public class MoviesCollection extends PagedCollection<MovieSearchResultResource> {
    public static List<Movie> toCollection(MoviesCollection response, Configuration config) {
        return response.toCollection(resource -> resource.toMovie(config));
    }
}

File: networking/src/main/java/com/jbrunton/mymovies/networking/resources/movies/MoviesCollection.kt
package com.jbrunton.mymovies.networking.resources.movies

import com.jbrunton.mymovies.entities.models.Configuration
import com.jbrunton.mymovies.entities.models.Movie
import com.jbrunton.mymovies.networking.resources.PagedCollection

class MoviesCollection(
        override val page: Int,
        override val total_results: Int,
        override val total_pages: Int,
        override val results: List<MovieSearchResultResource>
) : PagedCollection<MovieSearchResultResource>() {
    companion object {
        fun toCollection(response: MoviesCollection, config: Configuration): List<Movie> {
            return response.toCollection { resource -> resource.toMovie(config) }
        }
    }
}
